sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/ShuffleExchange.scala

```diff
@@ -40,14 +40,17 @@ case class ShuffleExchange(
     child: SparkPlan,
     @transient coordinator: Option[ExchangeCoordinator]) extends Exchange {
 
+  // NOTE: coordinator can be null after serialization/deserialization,
+  //       e.g. it can be null on the Executor side
+
```

Comment:
This note is meant to be a class internal comment, and not a comment on `override lazy val metrics = Map(`, so I'd say leaving the added empty line here makes more sense. Would you agree?